File: app/src/main/java/com/cn29/aac/ui/masterdetail/SimpleDetailActivityModule.java
package com.cn29.aac.ui.masterdetail;

import com.cn29.aac.ui.masterdetail.vm.SimpleMasterDetailShareViewModel;
import com.cn29.aac.ui.masterdetail.vm.SimpleMasterDetailShareViewModelFactory;

import androidx.lifecycle.ViewModelProviders;
import dagger.Module;
import dagger.Provides;

/**
 * Created by Charles Ng on 3/10/2017.
 */

@Module
public class SimpleDetailActivityModule {


  @Provides
  SimpleMasterDetailShareViewModel provideVm(SimpleDetailActivity simpleDetailActivity,
      SimpleMasterDetailShareViewModelFactory factory) {
    return ViewModelProviders
        .of(simpleDetailActivity, factory).get(SimpleMasterDetailShareViewModel.class);
  }
}

File: app/src/main/java/com/cn29/aac/ui/masterdetail/SimpleDetailActivityModule.kt
package com.cn29.aac.ui.masterdetail


import androidx.lifecycle.ViewModelProvider
import com.cn29.aac.ui.masterdetail.vm.SimpleMasterDetailShareViewModel
import com.cn29.aac.ui.masterdetail.vm.SimpleMasterDetailShareViewModelFactory
import dagger.Module
import dagger.Provides

/**
 * Created by Charles Ng on 3/10/2017.
 */
@Module
class SimpleDetailActivityModule {
    @Provides
    fun provideVm(simpleDetailActivity: SimpleDetailActivity,
                  factory: SimpleMasterDetailShareViewModelFactory): SimpleMasterDetailShareViewModel {
        return ViewModelProvider(simpleDetailActivity, factory)
                .get(
                        SimpleMasterDetailShareViewModel::class.java)
    }
}